### PR TITLE
Participation CSV exporting for CMS

### PR DESCRIPTION
**Pull request by grebneerg**
## Overview
Added support for exporting csv files for CMS containing participation grades.

## Changes Made
* Created general CSV retrieval route.
* Implemented CMS participation format.
* Expanded testing.

## Test Coverage
Added significant testing of several poll states and with different dates.

## Next Steps
Add support for canvas participation export.

### CODE REVIEW

File: src/routers/v2/CSVGenerator.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import User from '../../models/User';
+import constants from '../../utils/Constants';
+
+async function participation(id, dates: Array<Date>) {
+  const polls: Array<?Poll> = (await GroupsRepo.getPolls(id))
+    .filter((x: Poll) => {
+      const c = new Date(Number.parseInt(x.createdAt) * 1000);
+      return dates.some(d => c.toDateString() === d.toDateString());
+    })
+    .sort((a: Poll, b: Poll) => Number.parseInt(a.createdAt) - Number.parseInt(b.createdAt));
```
Comment (aliciaxw) [inline src/routers/v2/CSVGenerator.js]: using single letter variables is a bit hard to follow even if they're temp variables, maybe try something like `pollA` versus `a`

File: src/routers/v2/Groups/GetCSVRouter.js [hunk 2]
```diff
@@ -25,37 +22,30 @@ class GetCSVRouter extends AppDevRouter {
...
-      s.pipe(res);
+      const { format, dates } = req.query;
+
+      if (format === undefined || dates === undefined) {
+        res.sendStatus(400);
+        return;
+      }
+
+      const parsedDates = dates.map(Date.parse).map(n => new Date(n));
+      let s;
```
Comment (raahimenon) [inline src/routers/v2/Groups/GetCSVRouter.js]: `let s;`
please use an actual variable name my dude
